src/irq/irq_test.go

```diff
@@ -146,3 +149,235 @@ func mainLogicIRQ(t *testing.T, cfg IRQTestCase, i int) (string, error) {
 	}
 	return "", nil
 }
+
+func TestWriteCPUAffinitySuccessfulWrite(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	irqNum := 1
+	cpus := "0-3"
+	irqPath := filepath.Join(tmpDir, fmt.Sprintf("%d", irqNum))
+	if err := os.MkdirAll(irqPath, 0755); err != nil {
+		t.Fatalf("failed to create IRQ directory: %v", err)
+	}
+	affinityFile := filepath.Join(irqPath, "smp_affinity_list")
+	f, err := os.Create(affinityFile)
+	if err != nil {
+		t.Fatalf("failed to create file: %v", err)
+	}
+	f.Close()
+
+	procIRQ = tmpDir // override to avoid touching /proc
+	writer := &realIRQReaderWriter{
+		FileWriter: realFileWriter{},
+	}
+	err = writer.WriteCPUAffinity(irqNum, cpus)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	content, err := os.ReadFile(affinityFile)
+	if err != nil {
+		t.Fatalf("error reading back: %v", err)
+	}
+	if string(content) != cpus {
+		t.Errorf("expected %q, got %q", cpus, string(content))
+	}
+}
+
+type mockFileWriter struct{}
+
+func (mockFileWriter) WriteFile(path string, content []byte, perm os.FileMode) error {
+	return fmt.Errorf("input/output error") // Simulated /proc error
+}
+
+func TestWriteCPUAffinity_InputOutputErrorIgnored(t *testing.T) {
+	writer := &realIRQReaderWriter{
+		FileWriter: mockFileWriter{},
+	}
+	err := writer.WriteCPUAffinity(1, "0")
+	if err != nil {
+		t.Fatalf("expected nil, got error: %v", err)
+	}
+}
+
+// Simulate a real write error that's not ignorable (not "input/output error")
+func TestWriteCPUAffinityErrorReturned(t *testing.T) {
+	procIRQ = "/this/path/does/not/exist"
+
+	writer := &realIRQReaderWriter{
+		FileWriter: realFileWriter{},
+	}
+	err := writer.WriteCPUAffinity(99, "1-2")
+
+	if err == nil {
+		t.Fatal("expected an error but got nil")
+	}
+	if !strings.Contains(err.Error(), "error writing to") {
+		t.Errorf("unexpected error message: %v", err)
+	}
+}
+
+// Sanity: return nil even if file already has the value (line 63)
```

Comment:
Not easy a maintain line number in code.
```suggestion
// Sanity: return nil even if file already has the value
```